pkg/util/taints/taints_test.go

```diff
@@ -22,42 +22,90 @@ import (
 	"testing"
 
 	v1 "k8s.io/api/core/v1"
+
+	"github.com/stretchr/testify/assert"
 )
 
 func TestAddOrUpdateTaint(t *testing.T) {
-	node := &v1.Node{}
-
-	taint := &v1.Taint{
+	taint := v1.Taint{
 		Key:    "foo",
 		Value:  "bar",
 		Effect: v1.TaintEffectNoSchedule,
 	}
 
-	checkResult := func(testCaseName string, newNode *v1.Node, expectedTaint *v1.Taint, result, expectedResult bool, err error) {
+	taintNew := v1.Taint{
+		Key:    "foo_1",
+		Value:  "bar_1",
+		Effect: v1.TaintEffectNoSchedule,
+	}
+
+	taintUpdateValue := taint
+	taintUpdateValue.Value = "bar_1"
+
+	testcases := []struct {
+		name           string
+		node           *v1.Node
+		taint          *v1.Taint
+		expectedUpdate bool
+		expectedTaints []v1.Taint
+		taintsNum      int
```

Comment:
Is there a reason for having `taintsNum` as a field? Could we use `len(expectedTaints)` instead?

Reply:
ok, updated.